pkg/generators/models/enums.go

```diff
@@ -0,0 +1,178 @@
+package models
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/getkin/kin-openapi/openapi3"
+
+	tpl "github.com/contiamo/go-base/pkg/generators/templates"
+)
+
+// GenerateEnums outputs the Go enum models with validators
+func GenerateEnums(specFile io.Reader, dst string, packageName string) error {
+	data, err := ioutil.ReadAll(specFile)
+	if err != nil {
+		return fmt.Errorf("can not read spec file: %w", err)
+	}
+	swagger, err := openapi3.NewSwaggerLoader().LoadSwaggerFromData(data)
+	if err != nil {
+		return fmt.Errorf("can not parse the OpenAPI spec: %w", err)
+	}
+
+	enums := enumCtxs{}
+
+	// to do sort and iterate over the sorted schema
+	for name, s := range swagger.Components.Schemas {
+		if len(s.Value.Enum) == 0 {
+			continue
+		}
+
+		tctx := templateCtx{
+			SpecTitle:   swagger.Info.Title,
+			SpecVersion: swagger.Info.Version,
+			PackageName: packageName,
+			Name:        name,
+			VarName:     tpl.ToPascalCase(name),
+			Description: s.Value.Description,
+			Type:        s.Value.Type,
+			Values:      enumValues{},
+		}
+
+		for idx, v := range s.Value.Enum {
+			valstr := tpl.FirstUpper(fmt.Sprintf("%v", v))
+			valueVarName := tpl.ToPascalCase(fmt.Sprintf("%s %s", tctx.VarName, valstr))
+
+			if valueVarName == tctx.VarName {
+				valueVarName = tpl.ToPascalCase(fmt.Sprintf("%s Value %d", tctx.VarName, idx))
+			}
+
+			value := enumValue{
+				Name:    valueVarName,
+				VarName: valueVarName,
+				Value:   fmt.Sprintf("%q", v),
+			}
+			tctx.Values = append(tctx.Values, value)
+		}
+
+		sort.Sort(tctx.Values)
+		enums = append(enums, tctx)
+	}
+
+	sort.Sort(enums)
+	for _, tctx := range enums {
+		modelName := strings.ToLower(strings.ReplaceAll(fmt.Sprintf("model_%s.go", tpl.ToSnakeCase(tctx.Name)), " ", "_"))
+		filename := filepath.Join(dst, modelName)
+		fmt.Printf("writing %s\n", filename)
+		f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+		if err != nil {
+			return err
+		}
+
+		err = enumTemplate.Execute(f, tctx)
+		if err != nil {
+			return fmt.Errorf("failed to generate enum code: %w", err)
+		}
+
+		err = f.Close()
+		if err != nil {
+			return fmt.Errorf("failed to close output file: %w", err)
+		}
+	}
+	return nil
+}
+
+type templateCtx struct {
+	Filename    string
+	SpecTitle   string
+	SpecVersion string
+	PackageName string
+	Name        string
+	VarName     string
+	Type        string
+	Description string
+	Values      enumValues
+}
+
+type enumCtxs []templateCtx
+
+func (e enumCtxs) Len() int           { return len(e) }
+func (e enumCtxs) Swap(i, j int)      { e[i], e[j] = e[j], e[i] }
+func (e enumCtxs) Less(i, j int) bool { return e[i].Name < e[j].Name }
```

Comment:
Now I get, how you're using that `Sort` function, this implements the interface :)